helm/v1beta1/types_project_helm_chart_repository.go

```diff
@@ -0,0 +1,92 @@
+package v1beta1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:plural=projecthelmchartrepositories
+
+// ProjectHelmChartRepository holds namespace-wide configuration for proxied Helm chart repository
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type ProjectHelmChartRepository struct {
```

Comment:
I assume that `ProjectHelmChartRepository` is just a namespaced version of `HelmChartRepository`? If yes, we should keep them in sync, and to avoid large copy/paste blocks, we could do the following:

```golang
// +genclient
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
// +kubebuilder:plural=projecthelmchartrepositories

// ProjectHelmChartRepository holds namespace-wide configuration for proxied Helm chart repository
//
// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
// +openshift:compatibility-gen:level=2
type ProjectHelmChartRepository struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	// spec holds user settable values for configuration
	// +kubebuilder:validation:Required
	// +required
	Spec HelmChartRepositorySpec `json:"spec"`

	// Observed status of the repository within the namespace..
	// +optional
	Status HelmChartRepositoryStatus `json:"status"`
}

// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
// +openshift:compatibility-gen:level=2
type ProjectHelmChartRepositoryList struct {
	metav1.TypeMeta `json:",inline"`
	metav1.ListMeta `json:"metadata"`

	Items []ProjectHelmChartRepository `json:"items"`
}

With the above, we can even keep these two structs in `types_helm.go` file, along with the other types.
```